identity-account-storage/src/storage/stronghold.rs

```diff
@@ -335,69 +337,112 @@ impl Drop for Stronghold {
   }
 }
 
-async fn generate_private_key(vault: &Vault<'_>, location: &KeyLocation) -> Result<()> {
+fn generate_private_key(client: &Client, location: &KeyLocation) -> Result<()> {
   let generate_key: procedures::GenerateKey = procedures::GenerateKey {
     ty: location_key_type(location),
     output: location.into(),
-    hint: default_hint(),
   };
 
-  vault.execute(generate_key).await?;
+  client
+    .execute_procedure(generate_key)
+    .map_err(|err| procedure_error::<procedures::GenerateKey>(location, err))?;
 
   Ok(())
 }
 
-async fn retrieve_public_key(vault: &Vault<'_>, location: &KeyLocation) -> Result<PublicKey> {
-  let procedure: procedures::PublicKey = procedures::PublicKey {
-    ty: location_key_type(location),
-    private_key: location.into(),
-  };
+fn insert_private_key(client: &Client, mut private_key: PrivateKey, location: &KeyLocation) -> Result<()> {
+  let stronghold_location: Location = location.into();
+
+  let vault: ClientVault = client.vault(stronghold_location.vault_path());
+
+  let private_key_vec: Vec<u8> = private_key.as_ref().to_vec();
```

Comment:
Is there a way to leverage the `KeyStore` or Stronghold's [NC](https://github.com/iotaledger/stronghold.rs/blob/629466da83a677925904b2e733ef7bfad5a42864/engine/runtime/src/memories/noncontiguous_memory.rs#L83) types in favor of `PrivateKey` ? It somehow feels wasted to convert the private key to a vec, to just be zeroized and dropped afterwards.

Reply:
Appreciate the idea! Unfortunately, this would "leak" the stronghold dependency to our `Account` which is a no-go for Wasm. If you could factor out these NC types in a separate small crate that is also Wasm compatible, we could reconsider :wink:. Perhaps this would improve Wasm memory security, too.